weblate/checks/tests/test_same_checks.py

```diff
@@ -215,3 +215,8 @@ def test_same_uppercase(self):
         self.do_test(False, ("RMS", "RMS", ""))
         self.do_test(False, ("<primary>RMS</primary>", "<primary>RMS</primary>", ""))
         self.do_test(True, ("Who is RMS?", "Who is RMS?", ""))
+
+    def test_same_placeholders(self):
+        self.do_test(False, ("string $URL$", "string $URL$", "placeholders:$URL$"))
+        self.do_test(False, ("string $URL$", "string ", "placeholders:$URL$"))
+        self.do_test(True, ("string $URL$", "string $URL$", "placeholders:"))
```

Comment:
"string" is bad for testing as it's in same check blacklist. I'd probably just remove it as it's not really needed here and just adds confustion that it might be important.

Reply:
Should there be another word instead?

Reply:
I don't think any word is needed, this should test solely placeholders behavior.

Reply:
In this case, we want to test if the `same check` fires if placeholder is missing. It fires with `self.do_test(True, ("text $URL$", "text $URL$", "")` not with `self.do_test(True, ("$URL$", "$URL$", "")` I am unable to track down the problem, but i imagine, there is some other same check, that ignores specifict patteren like that.
I hope you can help me there.

Reply:
It might be that the placeholder is stripped out by other logic. I think there is something to trip out abbreviations, also it's likely that URL is in existing blacklist. So better choose something what works :-).